src/k8s-engine/images/imageProcessor.ts

```diff
@@ -504,6 +504,81 @@ export abstract class ImageProcessor extends EventEmitter {
 
     console.log(`Installing kim: kim ${ args.join(' ') }`);
 
+    // Need two loops here. First loop is to verify that we an successfully run `kim builder install`.
+    // If it fails because kubernetes is still running a docker node, we wait and retry, assuming that
+    // eventually it will stop using the docker node and will be running the newer containerd-based node.
+    // You can run `watch kubectl get node -o 'jsonpath={.items[].status.nodeInfo.containerRuntimeVersion}'`
+    // to see how this changes.
```

Comment:
You've checked that `kim builder install` failing due to wrong container runtime lets us try again (without `--force`), right?

If we know what we're looking for, wouldn't checking for that be useful?

Reply:
It does. If it fails and we don't retry, there's no subsequent attempt to install `buildkit`